src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -197,6 +197,31 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
     } yield ()
   }
 
+
+  //this class is getting to be too big. look into moving some things out of it.
+  def getPetServiceAccountKey(userId: WorkbenchUserId, project: GoogleProject): Future[ServiceAccountKey] = {
+    for {
+      maybePet <- directoryDAO.loadPetServiceAccount(PetServiceAccountId(userId, project))
+      response <- maybePet match {
+        case Some(pet) =>
+          googleIamDAO.createServiceAccountKey(project, pet.serviceAccount.email) recover {
+            case e: GoogleJsonResponseException if e.getDetails.getCode == StatusCodes.TooManyRequests.intValue => throw new WorkbenchException("You have reached the 10 key limit on service accounts. Please remove one to create another.")
+          }
+        case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Pet service account not found. Please create one first and then try again."))
```

Comment:
I think we should create the pet if it doesn't exist, call createUserPetServiceAccount instead of loadPetServiceAccount